### DISCUSSION TITLE
Registered and Trying to get a token via Postman - following ERCOT site/docs/support - same error - "error": "invalid_client", -

### DISCUSSION THREAD

**Discussion by BruceAltmann** [ercot/api-specs#35] [General]
I have asked ERCOT 2 times about this.
They send new Public API docs. (close to website)

Seems like my key is not really setup correctly.
I even tried the - formatting in my private key - I saw in their directions. No help.

Anyone else see this error
"error": "invalid_client",
    "error_description": "AADB2C90018: The client id '##privatekeyhere#######' specified in the request is not registered in tenant 'ercotb2c.onmicrosoft.com'.\r\nCorrelation ID: 73974027-9d77-4947-b932-d95f67fef61f\r\nTimestamp: 2024-06-18 21:57:51Z\r\n"

**Comment by mackermann-ercot** [comment]
Hello Bruce, 

Reviewing your registration, it appears that you have two subscriptions; "Public API" and "Public API2".  The first is in a cancelled state.  Are you using the subscription keys in API or API2 ?

Regards,
mackermann, ERCOT

**Reply by BruceAltmann** [reply]
yes
when it did not work
tried regen key on first one
same error
cancelled the first sub

created a new sub api2
so new keys
same error
so using api2 now


the docs all say use primary key
in 2 params

nothing works

**Comment by mackermann-ercot** [comment]
Hi Bruce, 

It seems you may be trying to use your subscription key in the call to get the Auth token.  From the documentation, use the actual client_id value that is in the instructions (also used in the scope parameter).  This client identification code identifies you as an ERCOT users to our authentication provider.  The subscription key is only used in subsequent calls to the API endpoints.

Use these parameters in a POST call **exactly** (only replacing $username and $password values):

grant_type:  password
username: $username
password: $password
scope: openid+fec253ea-0d06-4272-a5e6-b478baeecd70+offline_access
response: id_token
client_id: fec253ea-0d06-4272-a5e6-b478baeecd70

So your PostMan request should look like the following:

![ghs_2024-06-20_01](https://github.com/ercot/api-specs/assets/94029379/3fab5dab-30e9-4943-8ba4-45c24ddc6b48)


Regards,
Michael

**Reply by BruceAltmann** [reply]
ok

1) tried client_id fec253ea-0d06-4272-a5e6-b478baeecd70
got a token
thank you

I see 
the client_id is your api ID
I assumed it was my pkey for a subscription.


2) Please - make that clear in the the website/docs.

The client_id  is as shown below
It is the same client_id and scope for every user and subscription.
It is not your Primary Key.

**Comment by BruceAltmann** [comment]
thank for the help
